crud/src/main/java/com/redhat/lightblue/rest/crud/cmd/FindCommand.java

```diff
@@ -119,32 +119,40 @@ public void write(OutputStream os) throws IOException {
                         writer.write(chunkNode.toString());
                     }
                     writer.flush();
-                } finally {
+                } catch(Exception e) {
+                    metricCtx.markRequestException(e);
+                }
+                finally {
                     streamResponse.documentStream.close();
+                    metricCtx.endRequestMonitoring();
                 }
             }
         };
     }
 
     @Override
     public CallStatus run() {
+    	metricCtx = metrics.startStreamingEntityRequest("find", entity, version);
```

Comment:
We only want to start the streaming version if `stream` is true. Otherwise, use the regular version.

Also, slight nit pick, there is a mix of tabs and spaces in some of the changes it looks like